### ISSUE TITLE
Off-by-one error in RandomNumericSequenceGenerator?

### ISSUE THREAD

**Issue by dhilgarth** [AutoFixture/AutoFixture#148]
I have the feeling that there is a off-by-one error in `RandomNumericSequenceGenerator`'s [`CreateRange`](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/RandomNumericSequenceGenerator.cs#L187) method.

Consider an instance with minimum 0 and maximum 2. Asking an instance of `RandomNumericSequenceGenerator` three times to create a value will result in three values with the last value _always_ being `2`, although the maximum value should be an exclusive upper bound, so it actually should _never_ return 2.

The problem seems to be the check for the remaining limits:

``` C#
var remaining = this.limits.Where(x => x > this.upper - 1).ToArray();
```

With `limits = {0, 2}`, this will return `2`, because `2 > (2 - 1)`.

This also happens with more than two limits.

Consider the default constructor that uses these limits: 1, 255, 32767, ...

The first call to `CreateRange` will set lower to 1 and upper to 255, so it will return values that are greater than or equal to 1 and less than or equal to _254_.
At the next call `remaining` will be `{ 255, 32767, ...}`. This will result in `lower` to be set to 255 and `upper` to be set to _256_ (255 + 1).
In other words: After the first 254 random values that are less than or equal to 254, you will _always_ get the value 255.
I think, both the -1 in the `Where` clause as well as the +1 when assigning the upper value need to be removed to be semantically equivalent to the way it works for the first limit.
